packages/medusa/src/services/customer.ts

```diff
@@ -520,8 +525,8 @@ class CustomerService extends BaseService {
    *   castable as an ObjectId
    * @return {Promise} the result of the delete operation.
    */
-  async delete(customerId) {
-    return this.atomicPhase_(async (manager) => {
+  async delete(customerId: string): Promise<unknown> {
```

Comment:
question: softRemove is supposed to return the entity and not unknown?
`softRemove<T extends DeepPartial<Entity>>(entity: T, options?: SaveOptions): Promise<T & Entity>;`

Reply:
I had initially typed it as unknown since it's either the result of `softremove` or just an empty return, I have now typed it as `Promise<Customer | void>`